### ISSUE TITLE
entityManager.assign does not wrap Entity when wrappedReference = true

### ISSUE THREAD

**Issue by parisholley** [mikro-orm/mikro-orm#3092]
**Describe the bug**
https://github.com/mikro-orm/mikro-orm/blob/757801e9260b4cc89577b89ddbeec7a428fdb9a0/packages/core/src/entity/EntityAssigner.ts#L81

https://github.com/mikro-orm/mikro-orm/blob/757801e9260b4cc89577b89ddbeec7a428fdb9a0/packages/core/src/entity/EntityAssigner.ts#L134-L137

It would appear the default behavior is to just set the entity directly to the property, however if your entities make use of references like so:

```typescript
@ManyToOne(() => MikroEntity, { wrappedReference: true })
property?: IdentifiedReference<MikroEntity>;
```

you will get runtime errors trying to do things like `property.get()`

**Expected behavior**
I would expect `manager.assign()` to always inject a reference if `wrappedReference = true`

**Comment by B4nan** [comment]
Right, it should do `entity[prop.name] = Reference.wrapReference(value, prop);` instead
